### ISSUE TITLE
Unable to install it on centOS 7. Unable to install "start_gui_installer" and start "qmaster"

### ISSUE THREAD

**Issue by cloudliu01** [grisu48/gridengine#2]
Hi there,

Thanks for sharing the package!

I followed your guide to install it on CentOS 7.6 (1810)
`$git clone https://github.com/grisu48/gridengine.git`
`$cd gridengine/sge-8.1.9/source`
`$./scripts/bootstrap.sh`
`$./aimk -no-herd # also tried "-no-java" option but got the same result`

`# export SGE_ROOT="/opt/sge/"   ## Or whereever you want to install the grid engine to`
`# mkdir /opt/sge/ ## create target directory`
`# scripts/distinst -local -allall -noexit ## asks for confirmation`
`# cd $SGE_ROOT`

`[root@host1 source]# scripts/distinst -local -allall -noexit`

    No architecture specified. Assuming architecture is lx-amd64

    Installing: sge_qmaster sge_execd sge_shadowd sge_shepherd sge_coshepherd qstat qsub qalter qconf qdel qacct qmod qsh utilbin jobs qmon qhost qmake qtcsh qping qloadsensor.exe sgepasswd qquota qrsub qrstat qrdel qevent sge_share_mon distcommon arco sgeinspect
    Architectures: lx-amd64
    Base directory: /opt/sge
    OK [Y/N][Y]: y

    Installing "inst_sge", "install_qmaster" and "install_execd"
    Installing "util/" directory tree
    Installing "examples"
    Copying Pixmaps and Qmon resource file
    changing to /opt/sge/qmon/PIXMAPS
    ln -sf intro-sge.xpm intro.xpm
    ln -sf logo-sge.xpm logo.xpm
    Installing "pvm"
    Installing include files
    Installing drmaa.jar
    Installing juti.jar
    Installing jgdi.jar
    Installing JSV.jar
    Installing "Dtrace scripts"
    Installing "mpi/"
    Installing "hadoop/"
    Installing "man/"
    cp: cannot stat ‘MANSBUILD_sge/SEDMAN/man’: No such file or directory
    
    This command failed: cp -r MANSBUILD_sge/SEDMAN/man /opt/sge
    
    scripts/distinst: line 1166: cd: /opt/sge/man/man1: No such file or directory
    scripts/distinst: line 1172: cd: /opt/sge/man/man3: No such file or directory
    find: ‘/opt/sge/man’: No such file or directory
    chown: cannot access ‘/opt/sge/man’: No such file or directory
    
    This command failed: chown -R 0:0 /opt/sge/man
    
    Installing "doc/"
    Installing README, ... files
    common part done
    
    ARCo dir "/home/lliu/Downloads/sge-8.1.9/source/../arco/source" does not exist
    Installing "start_gui_installer" and "/util/gui-installer"
    cp: cannot stat ‘/home/lliu/Downloads/sge-8.1.9/source/clients/gui-installer/dist/installer.jar’: No such file or directory
    
    This command failed: cp /home/lliu/Downloads/sge-8.1.9/source/clients/gui-installer/dist/installer.jar /opt/sge/util/gui-installer
    
    GUI installer part done

Looks like the file installer.jar is missing. 

Then I tried to install the package
`# ./inst_sge -m -x -csp `
All seem OK except the last step. It was unable to start the qmaster daemon:
    Grid Engine qmaster startup
    ---------------------------

    Starting qmaster daemon. Please wait ...
       Starting Grid Engine qmaster
    
    sge_qmaster start problem
    
    sge_qmaster didn't start!
    sge_qmaster start problem

Any ideas?

Thanks
Liu

**Comment by grisu48** [comment]
Someone is really using this, nice :-)

Hi Liu,
I know that the installation is a not-straightforward process. I notices this error sometimes occuring, when something was wrong in the sge_master installation. Perhaps you missed something and made a typo somewhere. I found it especially helpful to check the following

* Does the username exists, under which the SGE should run?
* Are all the paths correctly set?
* Does the assigned user has permissions to access the SGE and SPOOL directories?

Unfortunately the SGE does not catch mistakes when they happen but only when you try to start the service. Please also check the systemd logs, perhaps there is a clue about what went wrong.


PS. The start_gui_installed I never checked. Could very well be that that one is not supported.

**Comment by grisu48** [comment]
@Minato4869 Could you check, if the _start_guit_installer_ is build?

**Comment by cloudliu01** [comment]
Hi Grisu48,

I believe I followed your steps, including the permission settings (build with SGE admin account 'lliu' and install with 'root' account). All paths were also the same as you suggested. Anyway, I will try it again tomorrow

Could you please tar your $SGE_ROOT folder and put it on Github? I guess I can use it directly, at least it should work if I create an admin user as yours? I will check the systemd log tomorrow

Many thanks for the quick reply

Best regards
Liu

**Comment by grisu48** [comment]
The one who asks shall receive :-) Please keep me posted, if you found something, then I can include that in the documentation.

[sge.tar.gz](https://github.com/grisu48/gridengine/files/3496746/sge.tar.gz)

**Comment by cloudliu01** [comment]
Hi Felix,

Thanks for providing the tar file. Unfortunately even with your version it still doesn't work - it is not able to start the qmaster service

I will need more time to debug. 

thanks for the help
best regards
Liu

**Comment by grisu48** [comment]
Can we close this issue?

**Comment by cloudliu01** [comment]
Hi Felix,

Sorry I am still not able to get it working in my environment. I think it might be OK to close it for now if no one else have seen the same issue

I really appreciate your time and efforts on my case!

Best regards
Liu

**Comment by cloudliu01** [comment]
Hi Felix,

Sorry for the late reply!

I managed to compile sge now. The issue was that some packages were missing in my previous installations. I had to specify the platform specific versions to get it working:
  java-1.8.0-openjdk-devel.x86_64
  java-1.8.0-openjdk.x86_64
  openssl.x86_64
  openssl-devel.x86_64
  libdbi-drivers.x86_64

Now I am working on installing it on my hosts. 

One tiny issue: ./util/gui-installer/installer.jar is still missing in my built. 

Thanks for the help!
Best regards
Liu
